### PR TITLE
Fix default value for keys during upgrade

### PR DESCRIPTION
**Pull request by hari-kuriakose**
## What

Ensure default value for  `DEFAULT_AUTH_USERNAME`, `DEFAULT_AUTH_PASSWORD`, `SYSTEM_ADMIN_USERNAME`, `SYSTEM_ADMIN_PASSWORD` keys during upgrade too.

## Why

Default value is required.

## How

For upgrade path, the `merge_env.py` script is invoked. Keys which always require a default value are marked by `SET_DEFAULT_KEYS` and a default value will be set for them if not found.

## Notes on Testing

Able to upgrade successfully.

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

**Review by ritwik-g — APPROVED** [review]

**Review by chandrasekharan-zipstack — APPROVED** [review]
LGTM, in case of ValueError due to these missing keys arising from `settings/base.py` we could suggest the user to run this script. I don't think there's a correlation between the two currently

**Review by kirtimanmishrazipstack — APPROVED** [review]
LGTM

File: docker/scripts/merge_env.py [hunk 1]
```diff
@@ -14,6 +14,13 @@
     "STRUCTURE_TOOL_IMAGE_URL",
     "STRUCTURE_TOOL_IMAGE_TAG",
 ]
+SET_DEFAULT_KEYS = [
+    "DEFAULT_AUTH_USERNAME",
+    "DEFAULT_AUTH_PASSWORD",
+    "SYSTEM_ADMIN_USERNAME",
+    "SYSTEM_ADMIN_PASSWORD",
```
Comment (ritwik-g) [inline docker/scripts/merge_env.py:21]: @hari-kuriakose we let's comment these out for now

**Review by ritwik-g — CHANGES_REQUESTED** [review]

File: docker/scripts/merge_env.py (hunk 1 above)
Comment (hari-kuriakose) [reply docker/scripts/merge_env.py:21]: @ritwik-g Why? This is required for someone having an existing setup and doing an upgrade.

We will need to have all of these, but use different default creds for default auth and system admin instead, as per the observed issue.

Comment (hari-kuriakose) [reply docker/scripts/merge_env.py]: @ritwik-g Patched this by setting different default creds for default and admin user.

**Review by nehabagdia — APPROVED** [review]

**Review by chandrasekharan-zipstack — APPROVED** [review]

File: docker/scripts/merge_env.py [hunk 2]
```diff
@@ -74,6 +82,10 @@ def _merge_to_env_file(
             if key not in PREFERRED_BASE_ENV_KEYS and key in target_env:
                 value = target_env.get(key, value)
 
+            # Set default value for these keys always.
+            if not value and key in SET_DEFAULT_KEYS:
+                value = SET_DEFAULT_KEYS[key]
```
Comment (chandrasekharan-zipstack) [inline docker/scripts/merge_env.py:87]: @hari-kuriakose usually people might not change this but will their values get overwritten whenever they merge env?

**Review by ritwik-g — COMMENTED** [review]
@hari-kuriakose we currently have commented out default admin values in run shell file. Shouldn't we reflect this change there as well?

File: docker/scripts/merge_env.py (hunk 2 above)
Comment (hari-kuriakose) [reply docker/scripts/merge_env.py:87]: @chandrasekharan-zipstack No, only when a user has an existing setup and this key was added in a later release. In this case, for example if `SYSTEM_ADMIN_USERNAME` is newly added in base nev (`sample.env`) but still empty at this point, then we need to set a default value as this key is marked by `SET_DEFAULT_KEYS` list.

For new setup, it is handled in the `run-platform.sh` script itself as `.env` is created fresh from `sample.env` for the very first time.

**Comment by hari-kuriakose** [comment]
> @hari-kuriakose we currently have commented out default admin values in run shell file. Shouldn't we reflect this change there as well?

Yes. We need to handle that separately.

Either default and admin credentials would be set in `backend.env` (ONLY if not flagged by Sonar) or they would be handled in the `run-platform.sh` script for [new installs](https://github.com/Zipstack/unstract/blob/1b3ba59a30dd7e71898bf19b30350497a63aabfb/run-platform.sh#L154).

**Review by ritwik-g — APPROVED** [review]

**Pull request merged**